parser/src/parser.rs

```diff
@@ -73,6 +73,50 @@ pub fn parse_expression(source: &str) -> Result<ast::Expression, ParseError> {
     do_lalr_parsing!(source, Expression, StartExpression)
 }
 
+pub fn parse(source: &str, mode: Mode) -> Result<ast::Top, ParseError> {
+    Ok(match mode {
+        Mode::Exec | Mode::Single => {
+            let ast = parse_program(source)?;
+            ast::Top::Program(ast)
+        }
+        Mode::Eval => {
+            let statement = parse_statement(source)?;
+            ast::Top::Statement(statement)
+        }
+    })
+}
+
+// compile mode but defined here
+#[derive(Clone, Copy)]
+pub enum Mode {
```

Comment:
I'm not too sure that this `Mode` enum should be contained in the parser crate. I feel it has its place in the compiler crate (where it was). The new `parse` function can be place in near the compile function in the builtins.rs in the vm crate.

Reply:
It seems `Mode` should be defined out of `compile.rs` because `ast.parse` works without `#[cfg(feature = "rustpython-compiler")]`. Any idea?

Reply:
@windelbouwman any advice for proper place about this?

Reply:
ast.parse should not work without the compiler crate, since when you have the compiler crate, the parser crate is implied. Vice versa is not true, but I think it makes sense to excluse `ast.parse` when the compiler feature is disabled. So I think we should place `Mode` in `mode.rs` in the `rustpython-compiler` crate.